torch/_refs/__init__.py

```diff
@@ -2189,6 +2196,7 @@ def amax(
         out=out,
         has_identity=False,
         output_dtype_kind=REDUCTION_OUTPUT_TYPE_KIND.SAME,
+        exact_dtype=True,
```

Comment:
Is there a reason that `amax` should be restricted this way? Or is this being done just to make the reference implementation of `amax` more consistent with the ATen implementation? 

cc @ngimel

Reply:
IMO, I don't think `amax` should have `exact_dtype=True`. See https://github.com/pytorch/pytorch/pull/87765#discussion_r1009531260 for the context for this flag.

Reply:
I did that just to match ATen. Being compatible makes testing easier, but I know we want refs to be "canonical". Since both of you think that the refs can be more flexible here, I'll change that back and add a comment in a test.

Reply:
So how would this work for low precision types? `_reduction` would convert input to fp32, produce fp32 result and then?

Reply:
Also, does `torch.amax` require the dtype of `out=` to exactly match that of the input? That'd be surprising.

Reply:
@lezcano It does.

```py
#!/usr/bin/env python3
import torch
a = torch.zeros(6, dtype=torch.int32).reshape(2,3)
out = torch.zeros(6, dtype=torch.int64).reshape(2,3)
torch.amax(a, out=out)
# RuntimeError: Expected the dtype for input and out to match, but got Int for input's dtype and Long for out's dtype.
```

Reply:
Ah, right, because it's a structured kernel. I think we've mostly been just generalising that behaviour in PrimTorch, but perhaps @mruberry has more context here.

Reply:
Also, what happens in the case pointed out by Natalia?

Reply:
I've rolled back the change that introduced `exact_dtype=True` on the refs side. LMK if there's anything left to discuss here.

Reply:
LGTM